runtime/lib/mirrors_impl.dart

```diff
@@ -1676,4 +1681,18 @@ class _Mirrors {
     }
     return typeMirror;
   }
+
+  static Type _resolveType(Type key, List<Type> typeArguments) {
+    if (!typeArguments.every((_) => _ is Type)) {
```

Comment:
I think you would would want to make sure that this is actually a '_Type' and not a user-defined subclass of `Type`.

Not sure if you have access to `_Type`, though.